pkg/crinit/init/init.go

```diff
@@ -120,11 +122,12 @@ func (o *initClusterRegistryOptions) Bind(flags *pflag.FlagSet, defaultServerIma
 	flags.BoolVar(&o.etcdPersistentStorage, "etcd-persistent-storage", true, "Use a persistent volume for etcd. Defaults to 'true'.")
 	flags.BoolVar(&o.dryRun, "dry-run", false, "Run the command in dry-run mode, without making any server requests.")
 	flags.StringVar(&o.apiServerOverridesString, "apiserver-arg-overrides", "", "Comma-separated list of cluster registry API server arguments to override, e.g., \"--arg1=value1,--arg2=value2...\"")
-	flags.StringVar(&o.apiServerServiceTypeString, apiserverServiceTypeFlag, string(v1.ServiceTypeLoadBalancer), "The type of service to create for the cluster registry. Options: 'LoadBalancer' (default), 'NodePort'.")
+	flags.StringVar(&o.apiServerServiceTypeString, apiserverServiceTypeFlag, string(v1.ServiceTypeNodePort), "The type of service to create for the cluster registry. Options: 'LoadBalancer', 'NodePort'.")
 	flags.StringVar(&o.apiServerAdvertiseAddress, apiserverAdvertiseAddressFlag, "", "Preferred address at which to advertise the cluster registry API server NodePort service. Valid only if '"+apiserverServiceTypeFlag+"=NodePort'.")
 	flags.Int32Var(&o.apiServerNodePortPort, apiserverPortFlag, 0, "Preferred port to use for the cluster registry API server NodePort service. Set to 0 to randomly assign a port. Valid only if '"+apiserverServiceTypeFlag+"=NodePort'.")
 	flags.BoolVar(&o.apiServerEnableHTTPBasicAuth, "apiserver-enable-basic-auth", false, "Enables HTTP Basic authentication for the cluster registry API server. Defaults to false.")
 	flags.BoolVar(&o.apiServerEnableTokenAuth, "apiserver-enable-token-auth", false, "Enables token authentication for the cluster registry API server. Defaults to false.")
+	flags.BoolVar(&o.apiServerStandalone, apiServerStandaloneFlag, false, "Disables the use of the Kubernetes API aggregation layer")
```

Comment:
I'm not sure what the right way to think about this flag is. I think there are potentially multiple things that are involved with standalone vs aggregated:

- service type
- authentication model
- necessary flags
- necessary backend setup
- more that I haven't thought of

I wonder if it would make sense to make two separate commands `initaggreated` and `initstandalone` (with better names) that could share logic but have separate flags as necessary.

Reply:
Yes, after I made this flag I started realizing there's going to be a bit of flag error checking and confusing list of possible arguments. I think limiting that with new subcommands would help.

The clusterregistry is going to have to deal with similar problems. We probably also want to think about how the clusterregistry should handle the differing arguments. Introducing new subcommands for an apiserver seems strange. So does maintaining 2 separate binaries.

Reply:
I opted for `crinit aggregated` and `crinit standalone` since the executable is already named `crinit`. Do we envision any other uses for this binary other than to initialize a CR? If that time comes we could then modify the command names too.